doc/source/serve/deploy_model_tutorial.rst

```diff
@@ -0,0 +1,327 @@
+function into a Ray Serve deployment that can be launched locally on a laptop.
+
+We start by opening a new Python file. First, we need to import ``ray`` and
+``ray serve``, to use features in Ray Serve such as ``deployments``, which
+provide HTTP access to our model.
+
+.. literalinclude:: ../../../python/ray/serve/examples/doc/e2e_deployment.py
+    :language: python
+    :start-after: __import_start__
+    :end-before: __import_end__
+
+After these imports, we can include our model code from above. 
+We won't call our ``summarize`` function just yet though! 
+We will soon add logic to handle HTTP requests, so the ``summarize`` function 
+can operate on article text sent via HTTP request.
+
+.. literalinclude:: ../../../python/ray/serve/examples/doc/e2e_deployment.py
+  :language: python
+  :start-after: __local_model_start__
+  :end-before: __local_model_end__
+
+Ray Serve needs to run on top of a Ray cluster, so we connect to a local one:
+
+.. literalinclude:: ../../../python/ray/serve/examples/doc/e2e_deployment.py
+  :language: python
+  :start-after: __start_ray_cluster_start__
+  :end-before: __start_ray_cluster_end__
+
+The ``address`` parameter in ``ray.init()`` connects your Serve script to a
+running local Ray cluster. Later, we'll discuss how to start a local Ray
+cluster.
+
+.. note::
+
+  ``ray.init()`` connects to or starts a single-node Ray cluster on your
+  local machine,  which allows you to use all your CPU cores to serve
+  requests in parallel. To start a multi-node cluster, see
+  :ref:`serve-deploy-tutorial`.
+
+Next, we start the Ray Serve runtime:
+
+.. literalinclude:: ../../../python/ray/serve/examples/doc/e2e_deployment.py
+  :language: python
+  :start-after: __start_serve_start__
+  :end-before: __start_serve_end__
+
+.. note::
+
+  ``detached=True`` means Ray Serve will continue running even when the Python
+  script exits. If you would rather stop Ray Serve after the script exits, use
+  ``serve.start()`` instead (see :doc:`deployment` for details).
+
+Now that we have defined our ``summarize`` function, connected to a Ray
+Cluster, and started the Ray Serve runtime, we can define a function that
+accepts HTTP requests and routes them to the ``summarize`` function. We
+define a function called ``router`` that takes in a Starlette ``request``
+object [#f1]_:
+
+.. literalinclude:: ../../../python/ray/serve/examples/doc/e2e_deployment.py
+  :language: python
+  :start-after: __router_start__
+  :end-before: __router_end__
+
+``router`` uses the ``"txt"`` parameter in the ``request`` to find the article
+text to summarize. It then passes the article text into the ``summarize``
+function and returns the value. We also add the decorator ``@serve.deployment``
+to the ``router`` function to turn the function into a Serve ``Deployment``
+object.
+
+.. tip::
+  This routing function's name doesn't have to be ``router``. 
+  It can be any function name as long as the corresponding name is present in 
```

Comment:
i don't get this sentence. i guess what you are trying to say that "function names are used by Serve as path prefix" or something?

Reply:
Yeah this tip is clarifying that the actual name of the `router` function can be anything. It explains the relationship between the name of the function and the endpoint URL.